server/src/main/java/org/elasticsearch/discovery/SeedHostsResolver.java

```diff
@@ -98,11 +100,12 @@ public static TimeValue getResolveTimeout(Settings settings) {
                 .stream()
                 .map(hn -> (Callable<TransportAddress[]>) () -> transportService.addressesFromString(hn))
                 .collect(Collectors.toList());
-        final List<Future<TransportAddress[]>> futures;
+        final SetOnce<List<Future<TransportAddress[]>>> futures = new SetOnce<>();
         try {
-            futures = executorService.get().invokeAll(callables, resolveTimeout.nanos(), TimeUnit.NANOSECONDS);
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
+            cancellableThreads.execute(()  -> {
```

Comment:
Nit: no need for braces

```suggestion
            cancellableThreads.execute(() ->
```